### PR TITLE
Quick fix distance metric

### PR DESCRIPTION
**Pull request by yunfeiguo**
- [x] [RES-566](https://binatechnologies.atlassian.net/browse/RES-566)
- [x] Blocking PRs: ADD HERE IF ANY
* People
  - [x] Reviewers (>=1): ADD HERE
  - [x] Merger (>=1): ADD HERE
  - [x] FYI: @ulitskyb
- [x] Release note (if needed)
- [x] No code copied from outside Bina 
- [x] Tests (Unit, Workflow, ITL & Gherkin) are passing
- [x] PR labels, milestones & assignees

# Summary

* report 3',5', distance of true and test variants, plus length difference
* print summary stats to stderr in the end

### CODE REVIEW

**Comment by yunfeiguo** [comment]
@marghoob if no issues, I will merge it.

File: src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java [hunk 1]
```diff
@@ -710,6 +714,13 @@ public void run(String[] args) {
         if (!parseArguments(args)) {
             return;
         }
+        if (isGlobalMatching) {
+            globalMatching();
+        } else {
+            localMatching();
+        }
+    }
+    private void localMatching() {
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java]: Style: just add another newline before this line.

File: src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java [hunk 2]
```diff
@@ -811,7 +822,7 @@ public void setNumberOfTrueCorrect(EnumStatsRatioCounter<VariantOverallType> num
         // this is for the original variants
         // it stores the total length of the original variant in bases
         // Still check for validation of canonical full variants
-        ArrayList<Integer> validatedTotalLength = new ArrayList<>();
+        ArrayList<Integer> trueVariantTotalLength = new ArrayList<>();
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java]: Shouldn't this be of type `List`?

File: src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java [hunk 3]
```diff
@@ -1065,6 +1089,9 @@ public void setNumberOfTrueCorrect(EnumStatsRatioCounter<VariantOverallType> num
 
         // read through again and compute for the true variants
         int numRead2 = 0;
+            List<Integer> tpThreePrimeDistance = new ArrayList<>();
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java]: Shouldn't this and the following be of type `List<Integer>`?

File: pom.xml [hunk 4]
```diff
@@ -95,6 +95,11 @@
             <artifactId>htsjdk</artifactId>
             <version>2.7.0</version>
         </dependency>
+	<dependency>
+	    <groupId>org.jgrapht</groupId>
```
Comment (marghoob) [inline pom.xml]: What is this being used for?

File: src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java (hunk 3 above)
Comment (yunfeiguo) [inline src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java]: Sorry, but this is already `List<Integer>`, isn't it?

File: pom.xml (hunk 4 above)
Comment (yunfeiguo) [inline pom.xml]: this will be used for global matching.

File: src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java (hunk 3 above)
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java]: Sorry, this was a mistake.

**Comment by marghoob** [comment]
Looks good to me. 👍

**Comment by yunfeiguo** [comment]
Thanks! merging...

**Pull request merged**